content/articles/understanding-layer-2-blockchain/index.md

```diff
@@ -0,0 +1,100 @@
+
+#### State channels
+In the [state channels](https://www.talentica.com/blogs/state-channels-an-introduction-to-off-chain-transactions/), parties interact directly in the blockchain network. Blockchain users can perform transactions without involving the primary chains. The miners spend minimal time, leading to fast processing rates.
+
+State channels do not need the transactions to be validated by the layer-1 blockchain. This is because validations of the resource are done through the smart-contract mechanism. Upon successful completion of the transaction, the resulting state is stored on the primary layer.
+
+State channels protect the transaction details that are exchanged between different parties. However, the final transaction details are stored in the ledger to be accessed publicly to maintain records.
+
+#### Sidechains
+A [sidechain](https://komodoplatform.com/en/academy/blockchain-sidechain/) is a side blockchain linked to the primary chain using a two-way peg. We can take an analogy of a forest where the trees can act as sidechains, and the forest itself is the primary chain. They are designed to handle a large batch of transactions. 
+
+A sidechain assists the primary chain with validating various transactions in the blockchain. The primary chain has ample time to deal with security and resolve disputes.
+
+Sidechains are not the same as state channels. The reason is that they store the transactions records on a ledger that can be publicly accessed. It means that if the sidechain is attacked, it cannot affect the primary chain's operations. However, sidechains involve much time and a significant amount of effort to design and build.
+
+#### Rollups
+These are layer-2 protocols that perform computations off the primary chain. In this blockchain, transaction details transfer occurs after a given time interval. This assists in maintaining records.
+
+Also, rollups handle transactions without interfering with the primary layer. This leads to higher throughput at minimal transaction costs. Since transaction details of the rollups are stored on the primary layer, the rollup's security can be guaranteed.
+
+#### Optimistic rollups
+Optimistic rollups regard that every transaction performed on the blockchain is valid. However, in most cases, optimistic rollups take a considerate amount of time to confirm the transactions. This waiting period gives rollups ample time to resolve a dispute if a challenge arises.
+
+#### Zero-knowledge rollups
+They perform computations off the chain before submitting the primary chain's validity proof. They at times use the [smart contract](https://www.ibm.com/topics/smart-contracts) to hold the funds on the base layer. Funds are then released once the proof of validity has been submitted. The primary chain has also validated and confirmed the transaction.
```

Comment:
Seems like either the last two lines are one line interrupted by a full stop or the last line feels a bit out of place. Rephrase it to convey the intended message.